crates/red_knot_python_semantic/src/types.rs

```diff
@@ -649,26 +651,35 @@ impl<'db> Type<'db> {
                         },
                     )
             }
-            (Type::ClassLiteral(..), Type::Instance(InstanceType { class }))
-                if class.is_known(db, KnownClass::Type) =>
-            {
-                true
-            }
             (Type::ClassLiteral(self_class), Type::SubclassOf(target_class)) => {
-                self_class.class.is_subclass_of(db, target_class.class)
+                self_class.class.is_subclass_of_base(db, target_class.base)
             }
-            (Type::SubclassOf(self_class), Type::SubclassOf(target_class)) => {
-                self_class.class.is_subclass_of(db, target_class.class)
+            (Type::Instance(self_class), Type::SubclassOf(target_class))
```

Comment:
similarly here, maybe the captured variables inside `Type::SubclassOf` could be renamed `target_ty` rather than `target_class`, since the inner data wrapped by this variant is no longer a class

Reply:
Done